src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs

```diff
@@ -250,40 +231,26 @@ public Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, IEnumer
 
                         this.SetBlockHash(transaction, nextBlockHash);
 
-                        all.Sort(UnspentOutputsComparer.Instance);
-                        foreach (UnspentOutputs coin in all)
+                        foreach (UnspentOutputs unspentOutput in unspentOutputs)
                         {
-                            this.logger.LogTrace("Outputs of transaction ID '{0}' are {1} and will be {2} to the database.", coin.TransactionId, coin.IsPrunable ? "PRUNABLE" : "NOT PRUNABLE", coin.IsPrunable ? "removed" : "inserted");
-                            if (coin.IsPrunable) transaction.RemoveKey("Coins", coin.TransactionId.ToBytes(false));
-                            else transaction.Insert("Coins", coin.TransactionId.ToBytes(false), coin.ToCoins());
+                            this.logger.LogTrace("Outputs of transaction ID '{transactionId}' are {prunableState} and will be {action} to the database.", unspentOutput.TransactionId, unspentOutput.IsPrunable ? "PRUNABLE" : "NOT PRUNABLE", unspentOutput.IsPrunable ? "removed" : "inserted");
+                            if (unspentOutput.IsPrunable) transaction.RemoveKey(CoinsKey, unspentOutput.TransactionId.ToBytes(false));
+                            else transaction.Insert(CoinsKey, unspentOutput.TransactionId.ToBytes(false), unspentOutput.ToCoins());
+                        }
 
-                            if (originalOutputs != null)
+                        foreach (RewindData rewindData in rewindDataCollection)
+                        {
+                            int nextRewindIndex = 0;
+                            if (this.TryGetRewindIndex(transaction, out int currentRewindIndex))
```

Comment:
I thought we said that the index comes from the block height, so no longer we should have such a logic here

Reply:
I am not sure what benefit we add by using block height instead of an index. Rewind does not accept block height as a parameter, so we don't know what block to rewind and will just rewind the last item from the storage. That would also mean that height will need to be added to `RewindData` object as it doesn't have it atm or possible infer it from `OutputsToRestore` if there are any.